innofw/core/datamodules/lightning_datamodules/image_to_text.py

```diff
@@ -0,0 +1,114 @@
+from innofw.core.datamodules.lightning_datamodules.base import BaseLightningDataModule
+from innofw.constants import Stages
+from pathlib import Path
+from typing import Optional, Dict
+from torch.utils.data import Dataset, DataLoader
+
+import pandas as pd
+import numpy as np
+import os
+import re
+import PIL.Image as pil
+import torch
+
+class ImageToTextDatamodule(BaseLightningDataModule):
+    task = "image-to-text"
+    
+    @staticmethod
+    def split_sentence(sentence):
+        return re.findall("\w+", sentence.lower())
+
+    @staticmethod
+    def encode_sentence(words, sentence, length: int):
+        sentence = ImageToTextDatamodule.split_sentence(sentence)
+        sentence = [words.get(x, words.get('<?>')) for x in sentence]
+        if len(sentence) + 2 > length:
+            raise ValueError(f"Unable to encode sentence of size {len(sentence)}\
+                             in a sentence of target length {length}")
+        sentence = [words.get("<s>")] + sentence + [words.get('</s>')]
+        while len(sentence) < length: sentence.append(words.get('<pad>'))
+        return sentence
+
+    class ImageToTextDataset(Dataset):
```

Comment:
Why not to separate this class to a file and add methods `split_text` and `encode_sentence` to it?